src/cne/elector.rs

```diff
@@ -17,6 +39,316 @@ pub struct Elector {
 
 impl From<String> for Elector {
   fn from(html_string: String) -> Self {
-    todo!()
+
+    if verify_citizenship_error(&html_string) {
+      panic!("Invalid citizenship provided");
```

Comment:
This doesn't goes here, the `From` trait is not responsible for validating the provided value.
Before trying to create an `Elector` from an arbitrary value, first you must check if the provided HTML is a valid candidate.

I would move this validation to the `request` module, so I could check if the response is valid before getting here.

Reply:
Also notice that the `from` function signature from the `From` trait always returns `Self`, this means that there's no way to return `Result` and handle the error from the caller side.

Now imagine that your app is running on production and it suddenly panics (causing your application to stop its execution), just because an invalid search has been made, I think you could recover from that easily!